app/src/main/kotlin/compiler/intermediate_form/FunctionDetailsGenerator.kt

```diff
@@ -1,33 +1,250 @@
 package compiler.intermediate_form
 
-import compiler.ast.Function
 import compiler.ast.NamedNode
+import compiler.ast.Variable
 import compiler.common.intermediate_form.FunctionDetailsGeneratorInterface
+import compiler.common.reference_collections.ReferenceHashMap
+import java.lang.Exception
+
+enum class VariableLocationType {
+    MEMORY,
+    REGISTER
+}
+
+val argPositionToRegister = listOf(Register.RDI, Register.RSI, Register.RDX, Register.RCX, Register.R8, Register.R9)
+val caleeSavedRegistersWithoutRSP = listOf(Register.RBX, Register.RBP, Register.R12, Register.R13, Register.R14, Register.R15)
 
 data class FunctionDetailsGenerator(
+    val parameters: List<NamedNode>,
+    val variableToStoreFunctionResult: Variable?,
+    val functionLocationInCode: IntermediateFormTreeNode.MemoryAddress,
     val depth: ULong,
-    val vars: Map<NamedNode, Boolean>,
-    val parameters: List<Function.Parameter>
+    val variablesLocationTypes: Map<NamedNode, VariableLocationType>, // should contain parameters
+    val displayAddress: IntermediateFormTreeNode,
+    val createRegisterFor: (NamedNode) -> Register = { Register() } // for testing
 ) : FunctionDetailsGeneratorInterface {
 
-    override fun generateCall(args: List<IntermediateFormTreeNode>): FunctionDetailsGeneratorInterface.FunctionCallIntermediateForm {
-        return TODO()
+    private val variablesStackOffsets: MutableMap<NamedNode, ULong> = ReferenceHashMap()
+    private val variablesRegisters: MutableMap<NamedNode, Register> = ReferenceHashMap()
+    private var variablesTotalOffset: ULong = 0u
+    private val previousDisplayEntryRegister = Register()
+
+    init {
+        for ((variable, locationType) in variablesLocationTypes.entries) {
+            when (locationType) {
+                VariableLocationType.MEMORY -> {
+                    variablesStackOffsets[variable] = variablesTotalOffset
+                    variablesTotalOffset += memoryUnitSize
+                }
+
+                VariableLocationType.REGISTER -> {
+                    variablesRegisters[variable] = createRegisterFor(variable)
+                }
+            }
+        }
+    }
+
+    override fun genCall(
+        args: List<IntermediateFormTreeNode>,
+    ): FunctionDetailsGeneratorInterface.FunctionCallIntermediateForm {
+        // First, it moves arguments to appropriate registers (or pushes to stack) according to x86 call convention.
+        // Then, it adds call instruction to actually call a given function
+        // At the end it creates IFTNode to get function result
```

Comment:
On the other hand it documents what this function does in one place, so that reader wouldn't have to read the whole method. But moved as you suggested, since for not too long methods it might be cleaner.